src/search/landmarks/landmark_cost_partitioning_algorithms.cc

```diff
@@ -19,122 +19,150 @@ using namespace std;
 namespace landmarks {
 CostPartitioningAlgorithm::CostPartitioningAlgorithm(
     const vector<int> &operator_costs, const LandmarkGraph &graph)
-    : lm_graph(graph), operator_costs(operator_costs) {
+    : landmark_graph(graph), operator_costs(operator_costs) {
 }
 
-const unordered_set<int> &CostPartitioningAlgorithm::get_achievers(
-    const Landmark &landmark, bool past) const {
+static const unordered_set<int> &get_achievers(
+    const Landmark &landmark, const bool past) {
     // Return relevant achievers of the landmark according to its status.
-    if (past) {
-        return landmark.possible_achievers;
-    } else {
-        return landmark.first_achievers;
-    }
+    return past ? landmark.possible_achievers : landmark.first_achievers;
 }
 
-
 UniformCostPartitioningAlgorithm::UniformCostPartitioningAlgorithm(
     const vector<int> &operator_costs, const LandmarkGraph &graph,
-    bool use_action_landmarks)
+    const bool use_action_landmarks)
     : CostPartitioningAlgorithm(operator_costs, graph),
       use_action_landmarks(use_action_landmarks) {
 }
 
-double UniformCostPartitioningAlgorithm::get_cost_partitioned_heuristic_value(
-    const LandmarkStatusManager &lm_status_manager,
-    const State &ancestor_state) {
-    vector<int> achieved_lms_by_op(operator_costs.size(), 0);
-    vector<bool> action_landmarks(operator_costs.size(), false);
-
-    ConstBitsetView past =
-        lm_status_manager.get_past_landmarks(ancestor_state);
-    ConstBitsetView future =
-        lm_status_manager.get_future_landmarks(ancestor_state);
-
-    double h = 0;
-
-    /* First pass:
-       compute which op achieves how many landmarks. Along the way,
-       mark action landmarks and add their cost to h. */
-    for (const auto &node : lm_graph) {
+/* Compute which operator achieves how many landmarks. Along the way, mark
+   action landmarks and sum up their costs. */
+double UniformCostPartitioningAlgorithm::first_pass(
+    vector<int> &landmarks_achieved_by_operator,
+    vector<bool> &action_landmarks,
+    ConstBitsetView &past, ConstBitsetView &future) {
+    double cost_action_landmarks = 0;
```

Comment:
I stumbled over this name, I would suggest `action_landmarks_cost` or `cost_for_action_landmarks`